sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -3532,6 +3532,49 @@ class DataFrameSuite extends QueryTest
       }.isEmpty)
     }
   }
+
+  test("SPARK-41207: Fix BinaryArithmetic with negative scale") {
+    withSQLConf(SQLConf.LEGACY_ALLOW_NEGATIVE_SCALE_OF_DECIMAL_ENABLED.key -> "true") {
```

Comment:
I don't think we need to add new features for a legacy config, unless this is a regression. Can you clarify which commit caused the regression?

Reply:
There are external reasons to not support negative scale decimal values. ANSI SQL disallows it, which is why SPARK-30252 turned it off by default. Parquet does not support it, which by the way that appears to be a bug in `ParquetTable` where it says that they are supported. I'll file an issue for it. 

To me there are two choices we either need to support it fully and start to work through all of the issues and corner cases to make them work all the time, or we need to deprecate them and remove them.  It has been three years I think we can move from legacy to deprecated. Having a "solution" with land mines hidden in it and only a config to protect you is not a good solution at all.

Reply:
I'm not sure this fixes all issues. Officially supporting negative scale is really a hard problem. Do you really turn on this config in production?